lib/react_client.dart

```diff
@@ -245,40 +318,10 @@ ReactComponentFactory _registerComponent(ComponentFactory componentFactory, [Ite
     )])
   ]);
```

Comment:
Note that we _could_ skip the `createFactory` call and just deal with the React component class (return result of `React.createClass`), passing it in as the `type` argument to `createElement` in `ReactComponentFactoryProxy` methods.

This still works fine, just an idea that may or may not be cleaner.